sim/core/spell_result.go

```diff
@@ -97,64 +93,57 @@ func (spell *Spell) PhysicalCritCheck(sim *Simulation, attackTable *AttackTable)
-		// Return correct value if ever used for a physical spell.
+		// PseudoStats.BonusDamage is physical "spell power", just return that here.
+		// TODO: Do "MobTypeSpellPower" effects for physical exist? E.g. something like "Do x extra weapon damage against y type"?
+		// If yes then it needs to be handled here and for the multi school case below.
 		return spell.Unit.PseudoStats.BonusDamage
 	case stats.SchoolIndexArcane:
-		return spell.Unit.GetStat(stats.ArcanePower)
+		bonusDamage = spell.Unit.GetStat(stats.ArcanePower)
 	case stats.SchoolIndexFire:
-		return spell.Unit.GetStat(stats.FirePower)
+		bonusDamage = spell.Unit.GetStat(stats.FirePower)
 	case stats.SchoolIndexFrost:
-		return spell.Unit.GetStat(stats.FrostPower)
+		bonusDamage = spell.Unit.GetStat(stats.FrostPower)
 	case stats.SchoolIndexHoly:
-		return spell.Unit.GetStat(stats.HolyPower)
+		bonusDamage = spell.Unit.GetStat(stats.HolyPower)
 	case stats.SchoolIndexNature:
-		return spell.Unit.GetStat(stats.NaturePower)
+		bonusDamage = spell.Unit.GetStat(stats.NaturePower)
 	case stats.SchoolIndexShadow:
-		return spell.Unit.GetStat(stats.ShadowPower)
+		bonusDamage = spell.Unit.GetStat(stats.ShadowPower)
 	default:
 		// Multi school: Get best power choice available.
-		maxPower := 0.0
 		for _, baseSchoolIndex := range spell.SchoolBaseIndices {
 			var power float64
 
-			// TODO / NOTE: Not a bug, just really not a nice solution imho.
-			// Not having physical power with the other power stats makes this if-else required.
-			// Ignoring this case would result in bad return values if physical multi schools with a coef > 0
-			// are ever a thing, due to SpellPower being before ArcanePower in stats.
-			// Also, just having this loop or having the switch above is irrelevant in terms of performance.
-			// The jump table above saves some instructions for normal spells but loop only seems to
-			// cause the function to be inlined, making the whole SpellPower() call inline.
-			// Overall just not nice the way it is.
+			// TODO: If physical multi school spells with a coef > 0 emerge at any point this function or
+			// the way bonus damage stats are set up needs to be changed.
+			// SpellPower, MobTypeSpellPower and SpellDamage stats are a convinience to group bonus damage for
+			// non-physical schools and would need to be considered against physical here and obviously not applied
+			// if physical is chosen.
 			if baseSchoolIndex == stats.SchoolIndexPhysical {
 				power = spell.Unit.PseudoStats.BonusDamage
 			} else {
 				// School and stat indices are ordered the same way.
 				power = spell.Unit.GetStat(stats.ArcanePower + stats.Stat(baseSchoolIndex) - 2)
 			}
 
-			if power > maxPower {
-				maxPower = power
+			if power > bonusDamage {
+				bonusDamage = power
 			}
 		}
-		return maxPower
 	}
+
+	return bonusDamage +
+		spell.Unit.GetStat(stats.SpellPower) +
+		spell.Unit.PseudoStats.MobTypeSpellPower +
+		spell.Unit.GetStat(stats.SpellDamage)
```

Comment:
What's the difference between stats.SpellPower and stats.SpellDamage? SpellPower for "increases damage and healing done", and SpellDamage for "increases spell damage"?
For the latter, I don't know if effects of this kind actually exist, e.g. Wizard Oils do in fact increase healing done as well.

Reply:
Arcane Elixir from the top of my head but there are others as well

Reply:
There are no effects in the game, Arcane Elixir also just adds a bonus for all magical schools. It's just used as a convenience in the sim to not have to add a bonus for multiple schools.

Reply:
I think SpellPower (as in: a single SpellEffect affecting both damage and healing spells) isn't a thing in SOD: all "increases damage and healing done" spells have two effects "inside" (Mod Healing Done (X) and Mod Damage Done (X)).
Arcane Elixir and a few others indeed only have "Mod Damage Done (X)", but obviously, `stats.SpellPower` is redundant with setting both `stats.SpellDamage` and `stats.SpellHealing`. There's not many dynamic effects increasing both, so there wouldn't be many changes if `stats.SpellPower` were removed.

This is obviously not for _this_ PR, but it would help the pipeline quite a bit, if <school>Power were removed from stats, and merged with their physical counterpart (`PseudoStats.BonusDamage`). This would also make the "damage done" and "damage taken" pipeline more symmetric, going forward.